api/app/graphql/mutations/reset_password.rb

```diff
@@ -0,0 +1,31 @@
+module Mutations
+    class ResetPassword < BaseMutation
+      argument :phone, String, required: true
+      argument :email, String, required: true
+      argument :otp, String, required: true
+      argument :password, String, required: true
+      
+      field :message, String, null: false
+  
+      def resolve(phone:, email:, otp:, password:)
+        # Find the user based on their phone number or email address
+        user = User.find_by(phone: phone)
+        user ||= User.find_by(email: email)
```

Comment:
You can use a nested query instead of writing two declarations.
You can reference this `ActiveRecord::QueryMethods` [Docs](https://api.rubyonrails.org/v7.0.4.2/classes/ActiveRecord/QueryMethods.html#method-i-where)